app/src/main/java/com/bigyoshi/qrhunt/qr/FragmentScanner.java

```diff
@@ -72,28 +76,52 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         codeScanner = new CodeScanner(activity, scannerView);
 
         codeScanner.setCamera(CodeScanner.CAMERA_BACK);
-        codeScanner.setScanMode(ScanMode.PREVIEW);
+        codeScanner.setScanMode(ScanMode.CONTINUOUS);
         codeScanner.setAutoFocusMode(AutoFocusMode.SAFE);
         codeScanner.setFlashEnabled(false);
         codeScanner.setAutoFocusEnabled(true);
         codeScanner.setFormats(CodeScanner.ALL_FORMATS);
 
+        /*button = root.findViewById(R.id.start_scan);
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                codeScanner.setScanMode(ScanMode.PREVIEW);
+                camera.processQRCode();
+                button.setVisibility(View.GONE);
+                codeScanner.setScanMode(ScanMode.CONTINUOUS);
+            }
+        });*/
+
         codeScanner.setDecodeCallback(result -> activity.runOnUiThread(() -> {
             camera = new QrCodeProcessor(FragmentScanner.this, result.getText(), playerId);
-            camera.processQRCode();
+            /*button.setVisibility(View.VISIBLE);
+            final Handler handler = new Handler(Looper.getMainLooper());
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    button.setVisibility(View.GONE);
+                }
+            }, 2000);*/
             codeScanner.setScanMode(ScanMode.PREVIEW);
-            codeScanner.startPreview();
+            camera.processQRCode();
+            final Handler handler = new Handler(Looper.getMainLooper());
+            handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    codeScanner.setScanMode(ScanMode.CONTINUOUS);
+                }
+            }, 2000);
```

Comment:
are we processing every QR code that's read regardless of button press? I'm a little confused here

Reply:
Yes, Once the scanner finds it will prompt, but you can just hit cancel.